engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -191,46 +202,64 @@ public void processAggregationAsync(Event event) {
     }
 
     public void processAggregationSync(Event event) {
-        AggregationCommand aggregationCommand;
+
+        List<AggregationCommand> aggregationCommands = new ArrayList<>();
         Timer.Sample consumedTimer = Timer.start(registry);
 
         try {
             Action action = actionParser.fromJsonString(event.getPayload());
-            Map<String, Object> map = action.getEvents().get(0).getPayload().getAdditionalProperties();
-            aggregationCommand = objectMapper.convertValue(map, AggregationCommand.class);
+            for (com.redhat.cloud.notifications.ingress.Event actionEvent : action.getEvents()) {
+                aggregationCommands.add(objectMapper.convertValue(actionEvent.getPayload().getAdditionalProperties(), AggregationCommand.class));
+            }
         } catch (Exception e) {
             Log.error("Kafka aggregation payload parsing failed for event " + event.getId(), e);
             rejectedAggregationCommandCount.increment();
             return;
         }
 
-        Log.infof("Processing received aggregation command: %s", aggregationCommand);
-        processedAggregationCommandCount.increment();
+        final String bundle  = aggregationCommands.get(0).getAggregationKey().getBundle();
 
+        processedAggregationCommandCount.increment();
         try {
-            Optional<Application> app = applicationRepository.getApplication(aggregationCommand.getAggregationKey().getBundle(), aggregationCommand.getAggregationKey().getApplication());
-            if (app.isPresent()) {
-                String eventTypeDisplayName = String.format("%s - %s - %s",
-                        event.getEventTypeDisplayName(),
-                        app.get().getDisplayName(),
-                        app.get().getBundle().getDisplayName()
-                );
-                eventRepository.updateEventDisplayName(event.getId(), eventTypeDisplayName);
+            if (!featureFlipper.isSingleDailyDigestEnabled()) {
+                processAggregateEmailsByAggregationKey(aggregationCommands.get(0), event);
+            } else {
+                processBundleAggregation(aggregationCommands, event);
```

Comment:
What if the aggregator triggers a bundle aggregation right after its deployment, but the new engine hasn't been deployed yet? Some data could be lost with the current code.

Reply:
Well because we have the same feature flag on aggregator and engine, we can controle the enabling sequence to avoid this case.
In addition, we can check on database when aggregation are scheduled for each org id to find a "free slot".
But if you think it's still risky, I can rework this part ?

Reply:
ok